pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -2306,75 +2320,104 @@ def init_arc_param(arctype):
-                + model.v_C_TotalBeneficialReuse
-                + model.p_alpha_AnnualizationRate
-                * (
-                    model.v_C_DisposalCapEx
-                    + model.v_C_StorageCapEx
-                    + model.v_C_TreatmentCapEx
-                    + model.v_C_PipelineCapEx
-                )
-                + model.v_C_Slack
-                - model.v_R_TotalStorage
-                - model.v_R_TotalBeneficialReuse
-            )
+    ####### Minimum cost objective #######
+    model.v_Z = Var(
+        within=Reals,
+        units=model.model_units["currency"],
+        doc="Objective function variable - minimize cost [currency]",
+    )
 
-        model.CostObjectiveFunction = Constraint(
-            rule=CostObjectiveFunctionRule, doc="Cost objective function"
+    model.ObjectiveFunctionCost = Constraint(
+        expr=model.v_Z
+        == model.v_C_TotalSourced
+        + model.v_C_TotalDisposal
+        + model.v_C_TotalTreatment
+        + model.v_C_TotalReuse
+        + model.v_C_TotalPiping
+        + model.v_C_TotalStorage
+        + model.v_C_TotalTrucking
+        + model.v_C_TotalBeneficialReuse
+        + model.p_alpha_AnnualizationRate
+        * (
+            model.v_C_DisposalCapEx
+            + model.v_C_StorageCapEx
+            + model.v_C_TreatmentCapEx
+            + model.v_C_PipelineCapEx
         )
+        + model.v_C_Slack
+        - model.v_R_TotalStorage
+        - model.v_R_TotalBeneficialReuse,
+        doc="Objective function constraint - minimize cost",
+    )
 
-    # Define reuse objective function #
+    model.objective_Cost = Objective(
+        expr=model.v_Z, sense=minimize, doc="Objective function - minimize cost"
+    )
 
-    elif model.config.objective == Objectives.reuse:
+    model.objective_Cost.deactivate()
 
-        def ReuseObjectiveFunctionRule(model):
-            return model.v_Z == -(
-                model.v_F_TotalReused / model.p_beta_TotalProd
-            ) + 1 / 38446652 * (
-                model.v_C_TotalSourced
-                + model.v_C_TotalDisposal
-                + model.v_C_TotalTreatment
-                + model.v_C_TotalReuse
-                + model.v_C_TotalPiping
-                + model.v_C_TotalStorage
-                + model.v_C_TotalTrucking
-                + model.v_C_TotalBeneficialReuse
-                + model.p_alpha_AnnualizationRate
-                * (
-                    model.v_C_DisposalCapEx
-                    + model.v_C_StorageCapEx
-                    + model.v_C_TreatmentCapEx
-                    + model.v_C_PipelineCapEx
-                )
-                + model.v_C_Slack
-                - model.v_R_TotalStorage
-                - model.v_R_TotalBeneficialReuse
-            )
+    ####### Maximum reuse objective #######
+    model.v_Z_Reuse = Var(
+        within=Reals,
+        units=pyunits.dimensionless,
+        doc="Objective function variable - maximize reuse [dimensionless]",
```

Comment:
@JavalVyas2000 yes, this variable is dimensionless, as it is a ratio of the total amount of water reused to the total amount of flowback/produced water.

Reply:
I am still not convinced on this, I agree it is the ratio in the first part of the objective but not in the second part of the objective where the costs are added. Only way this can be dimensionless is if the factor 1/38446652 has the unit 1/kUSD. Also do you believe for consistency, we shall add units to other objective functions as well?

Reply:
I mistakenly took the deleted code to be the one that was existing. This comments are taken care in the current version of the code. Hence, resolved.